caffe2/operators/distance_op.cc

```diff
@@ -223,10 +223,27 @@ bool DotProductOp<float, CPUContext>::RunOnDevice() {
   return true;
 }
 
+vector<TensorShape> TensorInferenceForDotProduct(
+    const OperatorDef& /* def */,
+    const vector<TensorShape>& in) {
+  vector<TIndex> dims;
+  if (in[0].dims().size() > 0) {
+    dims.push_back(in[0].dims(0));
+  }
+  return vector<TensorShape>{CreateTensorShape(dims, in[0].data_type())};
+}
+
 OpSchema::Cost CostInferenceForDotProduct(
     const OperatorDef& def,
     const vector<TensorShape>& in) {
-  struct OpSchema::Cost c = PointwiseCostInference<1>(def, in);
+  std::vector<TensorShape> out = TensorInferenceForDotProduct(def, in);
+  uint64_t size_out = 1;
+  for (auto i = 0; i < out[0].dims().size(); ++i) {
+    size_out *= out[0].dims(i);
```

Comment:
how about adding this method to TensorShape ?

Reply:
I've been trying to find where TensorShape is defined.  I *thought* it was a protobuf generated class and I don't have much experience with those yet.

Reply:
You are right. May be a utility function then?